applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_newton_raphson_erosion_process_strategy.hpp

```diff
@@ -374,56 +333,98 @@ class GeoMechanicsNewtonRaphsonErosionProcessStrategy
     /// <summary>
     /// Checks the status of the tip element and checks if it should continue or stop growing
     /// </summary>
-    /// <param name="n_open_elements"> number of open pipe elements</param>
-    /// <param name="n_elements"> number of pipe elements</param>
-    /// <param name="max_pipe_height"> maximum allowed pipe height</param>
+    /// <param name="NumberOfOpenPipeElements"> number of open pipe elements</param>
+    /// <param name="NumberOfPipeELements"> number of pipe elements</param>
+    /// <param name="MaxPipeHeight"> maximum allowed pipe height</param>
     /// <param name="PipeElements"> vector of all pipe elements</param>
     /// <returns>tuple of grow bool and number of open pipe elements</returns>
-    std::tuple<bool, int> check_status_tip_element(unsigned int          n_open_elements,
-                                                   unsigned int          n_elements,
-                                                   double                max_pipe_height,
-                                                   std::vector<Element*> PipeElements)
+    template <typename FilteredElementsType>
+    std::tuple<bool, int> CheckStatusTipElement(unsigned int NumberOfOpenPipeElements,
+                                                unsigned int NumberOfPipeELements,
+                                                double       MaxPipeHeight,
+                                                const FilteredElementsType& rPipeElements)
     {
         bool grow = true;
         // check status of tip element, stop growing if pipe_height is zero or greater than maximum
         // pipe height or if all elements are open
-        if (n_open_elements < n_elements) {
-            Element* tip_element = PipeElements.at(n_open_elements - 1);
-            double   pipe_height = tip_element->GetValue(PIPE_HEIGHT);
+        if (NumberOfOpenPipeElements < NumberOfPipeELements) {
+            auto   tip_element = rPipeElements.at(NumberOfOpenPipeElements - 1);
+            double pipe_height = tip_element->GetValue(PIPE_HEIGHT);
 
-            if ((pipe_height > max_pipe_height + std::numeric_limits<double>::epsilon()) ||
+            if ((pipe_height > MaxPipeHeight + std::numeric_limits<double>::epsilon()) ||
                 (pipe_height < pipe_height_accuracy)) {
                 // stable element found; pipe length does not increase during current time step
                 grow = false;
                 tip_element->SetValue(PIPE_EROSION, false);
                 tip_element->SetValue(PIPE_ACTIVE, false);
-                n_open_elements -= 1;
+                NumberOfOpenPipeElements -= 1;
```

Comment:
The more natural way to decrement in C++ is to use `--`:
```suggestion
                --NumberOfOpenPipeElements;
```

Reply:
naturally.